### ISSUE TITLE
Requests connexion error BadStatusLine

### ISSUE THREAD

**Issue by Linuxine** [matrix-org/matrix-python-sdk#133]
Hi,

When I try to use your sdk via [navi](https://github.com/madmax28/navi), or when I try to launch the sample:

`$ PYTHONPATH=. python samples/SimpleChatClient.py`

```
I provide my host, the username and password, and I get the following error:
Traceback (most recent call last):
  File "samples/SimpleChatClient.py", line 84, in <module>
    main(host, username, password, room_id_alias)
  File "samples/SimpleChatClient.py", line 39, in main
    client.login_with_password(username, password)
  File "/matrix-python-sdk/matrix_client/client.py", line 179, in login_with_password
    token = self.login_with_password_no_sync(username, password)
  File "/navi/navi/navi/matrix-python-sdk/matrix_client/client.py", line 156, in login_with_password_no_sync
    "m.login.password", user=username, password=password
  File "/navi/navi/matrix-python-sdk/matrix_client/api.py", line 128, in login
    return self._send("POST", "/login", content)
  File "/navi/navi/navi/matrix-python-sdk/matrix_client/api.py", line 545, in _send
    verify=self.validate_cert
  File "/usr/lib/python2.7/site-packages/requests/api.py", line 57, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/lib/python2.7/site-packages/requests/sessions.py", line 475, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/lib/python2.7/site-packages/requests/sessions.py", line 585, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python2.7/site-packages/requests/adapters.py", line 453, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', BadStatusLine("''",))
```
Is there anything special that needs to be done in my matrix configuration to avoid this ? I have created a normal user to be used has bot. I can login via Riot without issue with this user. 

Thanks for your help !

**Comment by non-Jedi** [comment]
Hi @Linuxine. That traceback doesn't look like it's a bug in this sdk to me. Could you try sending some https requests from the repl using the requests library and see if you get a similar error?

**Comment by non-Jedi** [comment]
Closing until more information is provided.

**Comment by Linuxine** [comment]
Hi, sorry I have been a little busy, forgot to check my emails for github. The issue was that my attempt to connect to the matrix server were failling when using 
~~~
       verify=self.validate_cert
~~~
in matrix_client/api.py, ligne 546, in def _send

If I remember correctly, this was due to an issue checking my lLt'sencrypt certificate. I had to manually edit the CA.cert to add the root CA to make it work, but this was not a very good solution, since I would have needed to edit it on every certificate renewal ? For now, I have set this setting to "verify=False" to allow my bot to connect to my matrix using the api.
